jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/PathMappings.java

```diff
@@ -182,15 +191,25 @@ public MappedResource<E> getMatch(String path)
                 }
             }
 
-            if (mr.getPathSpec().matches(path))
-                return mr;
+            matchedPath = mr.getPathSpec().matched(path);
```

Comment:
I think we have a big inefficiency here.... even in the existing code.

Consider if we have 3 EXACT mappings, 3 PREFIX and the DEFAULT mappings, then we will:
1. iterate over this loop 7 times.
2. twice we will fire the `group != lastGroup` test and do an efficient Trie lookup
3. once we fire the `group != lastGroup` and do nothing
4. So all 7 iterations come to this line and call `matched(path)`, even though we know for 6 of them that they will not match because they failed in the Trie

We need to change this loop so that if an EXACT or PREFIX match fails in the Trie, then all iterations for the same group are skipped and we don't try the explicit match here for them all.   

This change will more than make up for the small amount of extra garbage

Reply:
I think the original loop without the Tries is more efficient (in terms of speed and memory).

I'll work with @lorban and mock up some jmh tests to see what is truely better.

Reply:
I think we've addressed your concern on this block of code.